sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt

```diff
@@ -0,0 +1,250 @@
+package com.mapbox.maps.viewannotation
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewTreeObserver
+import android.widget.FrameLayout
+import androidx.annotation.LayoutRes
+import androidx.asynclayoutinflater.view.AsyncLayoutInflater
+import com.mapbox.maps.*
+import com.mapbox.maps.plugin.ViewPlugin
+import java.util.concurrent.ConcurrentHashMap
+
+internal class ViewAnnotationManagerImpl(
+  private val mapView: MapView,
+  private val mapboxMap: MapboxMap = mapView.getMapboxMap()
+) : ViewAnnotationManager, ViewAnnotationPositionsListener {
+
+  init {
+    mapView.requestDisallowInterceptTouchEvent(false)
+    mapboxMap.setViewAnnotationPositionsUpdateListener(this)
+  }
+
+  private val asyncInflater by lazy {
+    try {
+      AsyncLayoutInflater(mapView.context)
+    } catch (e: NoClassDefFoundError) {
+      throw RuntimeException(
+        "Please add https://mvnrepository.com/artifact/androidx.asynclayoutinflater/asynclayoutinflater/1.0.0 dependency " +
+          "to your project to make use of asynchronous view inflation when adding view annotation!"
+      )
+    }
+  }
+
+  private val annotationMap = ConcurrentHashMap<String, ViewAnnotation>()
+  private val idLookupMap = HashMap<View, String>()
+
+  // structs needed for drawing, declare them only once
+  private val currentViewsDrawnMap = HashMap<String, ScreenCoordinate>()
+  private val positionDescriptorMap = HashMap<String, ViewPosition>()
+  private val idsToRepositionSet = HashSet<String>()
+  private val idsToAddSet = HashSet<String>()
+  private val idsToDeleteSet = HashSet<String>()
+
+  override fun addViewAnnotation(
+    @LayoutRes resId: Int,
+    options: ViewAnnotationOptions,
+    asyncInflateCallback: (View) -> Unit
+  ) {
+    validateOptions(options)
+    asyncInflater.inflate(resId, mapView) { view, _, _ ->
+      asyncInflateCallback.invoke(prepareViewAnnotation(view, options))
+    }
+  }
+
+  override fun addViewAnnotation(
+    @LayoutRes resId: Int,
+    options: ViewAnnotationOptions
+  ): View {
+    validateOptions(options)
```

Comment:
should we save `resId` and throw exception to have consistent logic with `addViewAnnotation(view: View,...`

Reply:
current example inflates same layout each time and this is totally valid logic as inflating same resId will result in a new actual view

Reply:
make sense